src/AutoRest.CSharp/Common/Output/Models/Types/ObjectType.cs

```diff
@@ -3,107 +3,72 @@
         private ObjectTypeConstructor? _initializationConstructor;
+        private ObjectSerialization[]? _serializations;
+        private ObjectTypeDiscriminator? _discriminator;
 
-        public ObjectType(ObjectSchema objectSchema, BuildContext context) : base(context)
+        public ObjectType(BuildContext context) : base(context)
         {
-            OjectSchema = objectSchema;
-            _typeFactory = context.TypeFactory;
-            _serializationBuilder = new SerializationBuilder();
-            _usage = context.SchemaUsageProvider.GetUsage(OjectSchema);
-
-            var hasUsage = _usage.HasFlag(SchemaTypeUsage.Model);
-
-            DefaultAccessibility = objectSchema.Extensions?.Accessibility ?? (hasUsage ? "public" : "internal");
-            Description = BuilderHelpers.CreateDescription(objectSchema);
-
-            DefaultNamespace = GetDefaultNamespace(objectSchema, context);
-            _sourceTypeMapping = context.SourceInputModel?.CreateForModel(ExistingType);
-
-            // Update usage from code attribute
-            if (_sourceTypeMapping?.Usage != null)
-            {
-                foreach (var usage in _sourceTypeMapping.Usage)
-                {
-                    _usage |= Enum.Parse<SchemaTypeUsage>(usage, true);
-                }
-            }
         }
 
-        protected ObjectSchema OjectSchema { get; }
-
-        public bool IsStruct => ExistingType?.IsValueType == true;
-        protected override string DefaultName => OjectSchema.CSharpName();
-        protected override string DefaultAccessibility { get; } = "public";
-        protected override string DefaultNamespace { get; }
-        protected override TypeKind TypeKind => IsStruct ? TypeKind.Struct : TypeKind.Class;
-
-        public string? Description { get; }
+        public ObjectTypeConstructor[] Constructors => _constructors ??= BuildConstructors().ToArray();
+        public virtual ObjectTypeProperty[] Properties => _properties ??= BuildProperties().ToArray();
 
         public virtual CSharpType? Inherits => _inheritsType ??= CreateInheritedType();
-
+        public ObjectTypeConstructor? SerializationConstructor => _serializationConstructor ??= BuildSerializationConstructor();
+        public CSharpType? ImplementsDictionaryType => _implementsDictionaryType ??= CreateInheritedDictionaryType();
+        public ObjectTypeConstructor InitializationConstructor => _initializationConstructor ??= BuildInitializationConstructor();
         public ObjectSerialization[] Serializations => _serializations ??= BuildSerializations();
+        public bool IsStruct => ExistingType?.IsValueType == true;
+        public virtual string? Description { get; protected set; }
+        public ObjectTypeDiscriminator? Discriminator => _discriminator ??= BuildDiscriminator();
 
-        public ObjectTypeProperty[] Properties => _properties ??= BuildProperties().ToArray();
+        public abstract bool IncludeSerializer { get; }
```

Comment:
I think this makes sense on SchemaObject type only. As well as IncludeDeserializer and IncludeConverter. And possibly AdditionalPropertiesProperty

Reply:
The problem is the value is used in places where its expecting an ObjectType which is why we need the property here.  Otherwise we have to cast in dozens of places.  Having it here allows us to treat SystemObjectType and SchemaObjectType identically from all the writers perspectives